ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -410,15 +409,21 @@ public void put(MessageWriteSet messageSetToWrite) throws StoreException {
             ArrayList<IndexEntry> indexEntries = new ArrayList<>(messageInfo.size());
             for (MessageInfo info : messageInfo) {
               FileSpan fileSpan = log.getFileSpanForMessage(endOffsetOfLastMessage, info.getSize());
-              IndexValue value = new IndexValue(info.getSize(), fileSpan.getStartOffset(), info.getExpirationTimeInMs(),
-                  info.getOperationTimeMs(), info.getAccountId(), info.getContainerId());
+              // Put from frontend should always use 0 as lifeVersion. (0 is the starting life version number for any data).
+              // Put from replication or recovery should use liferVersion as it's.
+              short lifeVersion = IndexValue.hasLifeVersion(info.getLifeVersion()) ? info.getLifeVersion() : (short) 0;
+              IndexValue value =
+                  new IndexValue(info.getSize(), fileSpan.getStartOffset(), IndexValue.FLAGS_DEFAULT_VALUE,
+                      info.getExpirationTimeInMs(), info.getOperationTimeMs(), info.getAccountId(),
+                      info.getContainerId(), lifeVersion);
               IndexEntry entry = new IndexEntry(info.getStoreKey(), value, info.getCrc());
               indexEntries.add(entry);
               endOffsetOfLastMessage = fileSpan.getEndOffset();
             }
             FileSpan fileSpan = new FileSpan(indexEntries.get(0).getValue().getOffset(), endOffsetOfLastMessage);
             index.addToIndex(indexEntries, fileSpan);
             for (IndexEntry newEntry : indexEntries) {
+              System.out.println("Handle new put Entry for key " + newEntry.getKey());
```

Comment:
Is this for debug purpose? If it's really worth logging, how about using logger.trace ?

Reply:
will remove, for debug purpose.